ui/admin/app/components/mapping-list/index.js

```diff
@@ -1,16 +1,35 @@
 import Component from '@glimmer/component';
 import { tracked } from '@glimmer/tracking';
 import { action } from '@ember/object';
+import { assert } from '@ember/debug';
 
 export default class MappingListComponent extends Component {
-  @tracked
-  newOptionKey;
-  @tracked
-  newOptionValue;
+  // =attributes
 
+  /**
+   * @type {string}
+   */
+  @tracked newOptionKey;
+
+  /**
+   * @type {string}
+   */
+  @tracked newOptionValue;
+
+  // =actions
+
+  /**
+   * If a new key value is entered and an addOption method was specified,
+   * calls addOption with the new key and value.  Resets key and value.
+   */
   @action
   addOption() {
-    if (this.args.addOption) {
+    assert(
+      '[boundary-admin-key-value-list-field] `@addOption` is required.',
+      this.args.addOption
+    );
+
+    if (this.newOptionKey) {
```

Comment:
Do we want to always require a key to be able to add? If so, shouldn't we be disabling the add button instead?